pkg/kubelet/volume/iscsi/iscsi.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iscsi
+
+import (
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/volume"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/mount"
+	"github.com/golang/glog"
+)
+
+// This is the primary entrypoint for volume plugins.
+func ProbeVolumePlugins() []volume.Plugin {
+	return []volume.Plugin{&ISCSIDiskPlugin{nil}}
+}
+
+type ISCSIDiskPlugin struct {
+	host volume.Host
+}
+
+var _ volume.Plugin = &ISCSIDiskPlugin{}
+
+const (
+	ISCSIDiskPluginName = "kubernetes.io/iscsi-pd"
+)
+
+func (plugin *ISCSIDiskPlugin) Init(host volume.Host) {
+	plugin.host = host
+}
+
+func (plugin *ISCSIDiskPlugin) Name() string {
+	return ISCSIDiskPluginName
+}
+
+func (plugin *ISCSIDiskPlugin) CanSupport(spec *api.Volume) bool {
+	if spec.ISCSIDisk != nil {
+		return true
+	}
+	exe := exec.New()
+	// see if iscsiadm is there
+	command := exe.Command("iscsiadm", []string{"-h"}...)
```

Comment:
I don't understand this function - by th etime you get here you have already proven that the api.Volume under consideration is NOT an iscsi (you returned true above).

did you mean to check for == nil above, and return false?

This says to me that you don't cover this in any tests...